File: archive/legacy-python/converter/terrain_converter/style_json.py
"""Generate a MapLibre style that references the terrain DEM source."""

from __future__ import annotations

import json
from pathlib import Path


def build_style(
    tiles_url: str,
    *,
    source_name: str = "terrain-dem",
    style_name: str = "Terrain DEM Style",
    scheme: str = "xyz",
    encoding: str = "mapbox",
    tile_size: int = 256,
) -> dict[str, object]:
    return {
        "version": 8,
        "name": style_name,
        "sources": {
            source_name: {
                "type": "raster-dem",
                "tiles": [tiles_url],
                "encoding": encoding,
                "tileSize": tile_size,
                "scheme": scheme,
            }
        },
        "terrain": {"source": source_name, "exaggeration": 1.0},
        "layers": [],
    }


def write_style(path: str | Path, style: dict[str, object]) -> None:
    path = Path(path)
    path.parent.mkdir(parents=True, exist_ok=True)
    path.write_text(json.dumps(style, indent=2) + "\n", encoding="utf-8")

File: kotlin/terrain-core/src/jvmMain/kotlin/com/terrainconverter/core/StyleJson.kt
package com.terrainconverter.core

import java.nio.file.Files
import java.nio.file.Path

fun buildStyle(
    tilesUrl: String,
    sourceName: String = "terrain-dem",
    styleName: String = "Terrain DEM Style",
    scheme: String = "xyz",
    encoding: String = "mapbox",
    tileSize: Int = 256,
): Map<String, Any?> = jsonObject(
    "version" to 8,
    "name" to styleName,
    "sources" to jsonObject(
        sourceName to jsonObject(
            "type" to "raster-dem",
            "tiles" to listOf(tilesUrl),
            "encoding" to encoding,
            "tileSize" to tileSize,
            "scheme" to scheme,
        ),
    ),
    "terrain" to jsonObject(
        "source" to sourceName,
        "exaggeration" to 1.0,
    ),
    "layers" to emptyList<Any>(),
)

fun writeStyle(path: Path, style: Map<String, Any?>) {
    path.parent?.let { Files.createDirectories(it) }
    Files.writeString(path, renderPrettyJson(style))
}
